R-package/inst/make-r-def.R

```diff
@@ -0,0 +1,61 @@
+# [description]
+#     Create a definition file (.def) from a .dll file, using objdump.
+#
+# [usage]
+#
+#     Rscript make-r-def.R something.dll something.def
+#
+# [references]
+#    * https://www.cs.colorado.edu/~main/cs1300/doc/mingwfaq.html
+
+args <- commandArgs(trailingOnly = TRUE)
+
+IN_DLL_FILE <- args[[1L]]
+OUT_DEF_FILE <- args[[2L]]
+DLL_BASE_NAME <- basename(IN_DLL_FILE)
+
+message(sprintf("Creating '%s' from '%s'", OUT_DEF_FILE, IN_DLL_FILE))
+
+# use objdump to dump all the symbols
+OBJDUMP_FILE <- "objdump-out.txt"
+exit_code <- system2(
+    command = "objdump"
+    , args = c(
+        "-p"
+        , shQuote(IN_DLL_FILE)
+    )
+    , stdout = OBJDUMP_FILE
+)
```

Comment:
ah! I need to change this to use `{processx}`. The MSYS job is taking 25+ minutes and I'm guessing that is the reason.

Reply:
Just  pushed https://github.com/microsoft/LightGBM/pull/3065/commits/8f28725e9a2f5b09e2712407942f84f60d0ec4f4, let's see if it helps the time.

You'll see I added a `.pipe_shell_command_to_stdout()` that looks similar to the `.run_shell_command()` we have in other scripts (https://github.com/microsoft/LightGBM/blob/3caf359a79754825aac17ac77d557d0b7288c70e/R-package/src/install.libs.R#L23). That is because `processx::run()` can't be used to redirect command stdout (https://github.com/r-lib/processx/issues/254)